build_modules/test/meta_module_test.dart

```diff
@@ -447,4 +447,21 @@ void main() {
           reason: meta.modules.map((m) => m.toJson()).toString());
     }
   });
+
+  test('libraries can import themselves via empty import', () async {
+    var assets = makeAssets({
+      'myapp|lib/a.dart': '''
+            import 'a.dart';
+          ''',
+    });
+
+    var a = AssetId('myapp', 'lib/a.dart');
+
+    var expectedModules = [
+      matchesModule(Module(a, [a], [], defaultPlatform, true)),
```

Comment:
@jakemac53 I made sure that this module importing itself does not have any dependencies. Does that clear your concerns? I didn't see any additional errors in manual testing either.

Reply:
Testing that it doesn't list itself in deps is sufficient yes :)